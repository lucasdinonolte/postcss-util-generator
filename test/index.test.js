const postcss = require('postcss');
const plugin = require('../src/index.js');

async function run(input, opts = {}) {
  let result = await postcss([plugin(opts)]).process(input, {
    from: undefined,
  });

  expect(result.css).toMatchSnapshot();
  expect(result.warnings()).toHaveLength(0);
}

const rootDecl = `:root {
  --color-blue: #0000ff;
  --color-blue-100: #0000ff;
}

@media (min-width: 640px) {
  :root {
    --color-blue: #0000ff;
  }
}`;

const input = `${rootDecl}

@utils;`;

const inputReverse = `@utils;

${rootDecl}`;

describe('postcss-util-generator', () => {
  it('works', async () => {
    await run(input, { staticUtilities: {} });
  });

  it('works with reverse', async () => {
    await run(inputReverse, { staticUtilities: {} });
  });

  it('generates static utilities', async () => {
    const staticInput = `@utils`;

    await run(staticInput, {
      staticUtilities: {
        text: {
          items: { left: 'left', right: 'right' },
          properties: ['text-align'],
        },
      },
    });
  });

  it('should only build the specified utilities', async () => {
    const input = `${rootDecl}; @utils(color);`;

    await run(input, {});
  });
});
