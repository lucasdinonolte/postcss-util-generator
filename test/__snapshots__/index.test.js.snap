// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss-util-generator generates static utilities 1`] = `
".textLeft {
    text-align: left
}
.textRight {
    text-align: right
}"
`;

exports[`postcss-util-generator should only build the specified utilities 1`] = `
":root {
  --color-blue: #0000ff;
  --color-blue-100: #0000ff;
}

@media (min-width: 640px) {
  :root {
    --color-blue: #0000ff;
  }
}; .textBlue {
  color: var(--color-blue);
}; .textBlue100 {
  color: var(--color-blue-100);
}"
`;

exports[`postcss-util-generator works 1`] = `
":root {
  --color-blue: #0000ff;
  --color-blue-100: #0000ff;
}

@media (min-width: 640px) {
  :root {
    --color-blue: #0000ff;
  }
}

.bgBlue {
  background-color: var(--color-blue);
}

.bgBlue100 {
  background-color: var(--color-blue-100);
}

.textBlue {
  color: var(--color-blue);
}

.textBlue100 {
  color: var(--color-blue-100);
}"
`;

exports[`postcss-util-generator works with reverse 1`] = `
".bgBlue {
  background-color: var(--color-blue);
}

.bgBlue100 {
  background-color: var(--color-blue-100);
}

.textBlue {
  color: var(--color-blue);
}

.textBlue100 {
  color: var(--color-blue-100);
}

:root {
  --color-blue: #0000ff;
  --color-blue-100: #0000ff;
}

@media (min-width: 640px) {
  :root {
    --color-blue: #0000ff;
  }
}"
`;
